sdk/python/feast/client.py

```diff
@@ -716,22 +729,25 @@ def ingest(
 
         # Update the feature set based on PyArrow table of first row group
         if force_update:
-            feature_set.infer_fields_from_pa(
-                table=pq_file.read_row_group(0),
-                discard_unused_fields=True,
-                replace_existing_features=True,
-            )
-            self.apply(feature_set)
+            if isinstance(feature_set, FeatureSet):
+                feature_set.infer_fields_from_pa(
+                    table=pq_file.read_row_group(0),
+                    discard_unused_fields=True,
+                    replace_existing_features=True,
+                )
+                self.apply(feature_set)
+            else:
+                raise Exception("Use feature set object to do force_update")
         current_time = time.time()
 
         print("Waiting for feature set to be ready for ingestion...")
         while True:
             if timeout is not None and time.time() - current_time >= timeout:
                 raise TimeoutError("Timed out waiting for feature set to be ready")
-            feature_set = self.get_feature_set(name, version)
+            feature_set_spec = self.get_feature_set(name, version)
```

Comment:
why the rename?

Reply:
The `feature_set` is a parameter already defined on the function arguments.

Mypy considers it is a bad practice when we reuse the same variable but assign it to a different type. We will lose track of what the `feature_set` type is. Hence the new variable creation

Reply:
The spec is used typically for the proto representation. Can you think of a different way to solve this? I think we should be fetching the feature set from Feast Core if they provide a string. That would also solve the exception above.

Reply:
I have tried to use the above solution actually. Even though I add the check for the `feature_set` to make sure that it was `FeatureSet` object, the linter won't acknowledge the new type, because the type was defined in the parameter, so reusing the same variable for different type was not allowed.

I think we can change the name? Are you happy if we add `_` in front of the variable?
 ```suggestion
            _feature_set = self.get_feature_set(name, version)
```

Reply:
Can you either use `cast` or `assert isinstance`?

https://github.com/python/mypy/issues/2589#issuecomment-268044680
https://mypy.readthedocs.io/en/stable/casts.html

Reply:
Cool! Thanks, the cast works like charm, will keep it notes